### PR TITLE
7tv emotes support

### PR DESCRIPTION
**Pull request by MagicHack**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable
- [x] Documentation in docs/ or install-docs/ was updated, if applicable

I saw it was discussed in  #1256 that support for 7tv may be too early, so idk.
The people I host the bot for and myself use 7tv so I added it eShrug.

### CODE REVIEW

File: pajbot/apiwrappers/seventv.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        """Returns a list of global 7TV emotes in the standard Emote format."""
+        query_string = """{
+            search_emotes(query: "", globalState: "only", limit: 150, pageSize: 150) {
+                id
+                name
+            }
+        }"""
+        params = {"query": query_string}
+        response = self.post("gql", json=params)
+        return self.parse_emotes(response["data"]["search_emotes"])
```
Comment (RAnders00) [inline pajbot/apiwrappers/seventv.py:43]: I don't like that we don't paginate here (download multiple pages if necessary)

**Review by RAnders00 — CHANGES_REQUESTED** [review]
I only see one problem, apart from that this looks good! Thank you!

**Comment by alazymeme** [comment]
Any plans to add support for the CLR modules, emote limit and I think you can add it to the quests module?

**Comment by zneix** [comment]
7tv REST API is just around the corner, no idea if you want to wait for it or go with gql as it is right now.

File: pajbot/apiwrappers/seventv.py (hunk 1 above)
Comment (MagicHack) [reply pajbot/apiwrappers/seventv.py]: I implemented the pagination by checking if we get a full page from the query.
There's  alse the `x-collection-size` header that could be checked for the total number of emotes, would require a post_response method to be added to the base api. Might be  a better approach.

**Comment by RAnders00** [comment]
> Any plans to add support for the CLR modules, emote limit and I think you can add it to the quests module?

CLR modules support all emote providers automatically through the unified `Emote` model that all providers create.

Emote limit module also supports all emote providers.

Quests module: https://github.com/MagicHack/pajbot/blob/228700d7c601246f96bd76231a64b8f3f6733688/pajbot/modules/quests/typeemote.py#L78 We currently only select from global Twitch emotes. Support for third-party emotes in general would definitely be a different PR.

**Review by RAnders00 — APPROVED** [review]

**Comment by RAnders00** [comment]
Please don't merge yet, waiting for @pajlada's review too

**Review by pajlada — APPROVED** [review]
Looks good to me - thanks for the clean addition @MagicHack

**Comment by pajlada** [comment]
will merge in after #1253

**Pull request merged**